src/index.ts

```diff
@@ -0,0 +1,99 @@
+import * as DCL from 'decentraland-ecs'
+const typesMap = require('decentraland-ecs/types/dcl/decentraland-ecs.api')
+
+type Component = {
+  data?: DCL.ObservableComponent
+  entityName?: string
+}
+
+export class SceneWriter {
+  private entities: Map<string, DCL.Entity> = new Map<string, DCL.Entity>()
+  private components: Component[] = []
+
+  addEntity(name: string, entity: DCL.Entity) {
+    this.entities.set(name, entity)
+  }
+
+  addComponent(entityName: string, component: DCL.ObservableComponent) {
+    this.components.push({ data: component, entityName })
+  }
+
+  emitCode(): string {
+    let code = ''
+    this.entities.forEach((entity, name) => {
+      code += this.writeEntity(name) + '\n'
+    })
+    return code
+  }
+
+  private writeEntity(name: string): string {
+    let code = ''
+    code += `const ${name} = new Entity()\n`
+    this.components
+      .filter(c => c.entityName === name)
+      .forEach(c => {
+        code += `${name}.set(${this.writeComponent(c.data)})\n`
+      })
+    return code
+  }
+
+  private writeComponent(component: DCL.ObservableComponent) {
+    // Get class name
+    let constructor
+    const typesArray = Object.keys(typesMap.exports).filter(
+      n => typesMap.exports[n].kind === 'class' && n !== 'Shape' && n !== 'ObservableComponent'
+    )
+    for (let i = 0; i < typesArray.length; i++) {
+      if (component instanceof DCL[typesArray[i]]) {
```

Comment:
```suggestion
      if (typeof DCL[typesArray[i]] === 'function' && component instanceof DCL[typesArray[i]]) {
```

Avoid making instanceof number / null / other things that we may be exporting

Reply:
I think that this case doesn't exist since typesArray could be only classes (we're filtering before by kind field == "Class") if a function came up at DCL[typesArray[i]] it means that ecs is wrong, not this file